src/core/pieces/PieceFactory.ts

```diff
@@ -12,7 +14,14 @@ class PieceFactory {
   public static createPiece(type: PieceType, color: Color, position: Position): Piece {
     switch(type) {
       case PieceType.Pawn:
-        return new Pawn(color, position);
+        const pawn = new Pawn(color, position);
+        if((position.x !== 6 && color === Color.White) || (position.y !== 1 && color === Color.Black)) {
+          const movePolicies = [new MovePolicy(MoveType.Vertical, new MoveConstraints({max: 1})),
+            new MovePolicy(MoveType.Diagonal, new MoveConstraints( {max: 1} ))];
+          pawn.updatePolicies(movePolicies)
+          pawn.afterFirstMove = true;
+        }
```

Comment:
I think this logic should be inside Pawn constructor - why should PieceFactory know such implementation details of Pawn piece?

Reply:
Makes sense, will put it in Pawn's constructor.